src/test/java/bio/terra/service/snapshotbuilder/SnapshotRequestDaoTest.java

```diff
@@ -0,0 +1,217 @@
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+
+import bio.terra.common.EmbeddedDatabaseTest;
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.NotFoundException;
+import bio.terra.common.fixtures.DaoOperations;
+import bio.terra.common.fixtures.ProfileFixtures;
+import bio.terra.common.fixtures.ResourceFixtures;
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.BillingProfileRequestModel;
+import bio.terra.model.SnapshotAccessRequest;
+import bio.terra.model.SnapshotAccessRequestResponse;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.profile.ProfileDao;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.resourcemanagement.google.GoogleResourceDao;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles({"google", "unittest"})
+@Tag(Unit.TAG)
+@EmbeddedDatabaseTest
+class SnapshotRequestDaoTest {
+
+  @Autowired private DaoOperations daoOperations;
+  @Autowired private ProfileDao profileDao;
+  @Autowired private GoogleResourceDao resourceDao;
+  @Autowired private SnapshotRequestDao snapshotRequestDao;
+
+  private BillingProfileModel billingProfile;
+  private GoogleProjectResource projectResource;
+  private Dataset dataset;
+  private SnapshotAccessRequest snapshotAccessRequest;
+
+  private SnapshotAccessRequestResponse response;
+  private String email;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    BillingProfileRequestModel profileRequest = ProfileFixtures.randomBillingProfileRequest();
+    billingProfile = profileDao.createBillingProfile(profileRequest, "testUser");
+
+    projectResource = ResourceFixtures.randomProjectResource(billingProfile);
+    UUID projectId = resourceDao.createProject(projectResource);
+    projectResource.id(projectId);
+
+    dataset = daoOperations.createMinimalDataset(billingProfile.getId(), projectId);
+
+    snapshotAccessRequest = SnapshotBuilderTestData.ACCESS_REQUEST;
+    email = "user@gmail.com";
+    response = snapshotRequestDao.create(dataset.getId(), snapshotAccessRequest, email);
+  }
+
+  @Test
+  void getById() {
+    assertThat(
+        "Snapshot Access Request should be the same as the example",
+        snapshotRequestDao.getById(response.getId()),
+        equalTo(response));
```

Comment:
Makes sense. So I've replaced this and other direct checks with a similar verify method. But what about the contains check for the enumerate method? That seems like another direct check that would fail if the mapper doesn't work. Should I replace that with calls to the verify method too?